### DISCUSSION TITLE
Advanced paste escape for NULL values

### DISCUSSION THREAD

**Discussion by radek-vytiska** [dbeaver/dbeaver#11427] [Q&A] [answered]
Hello,

I often use advanced paste from Excel in data edit mode. I prepare data structure in Excel, then I use advanced paste on new line in data editor and everything is pasted perfectly. After that I generate script and I have my SQL ready for deploy.

But there is one HUGE problem that prevents this worflow to be perfect. I didnt figured out, how to "escape" or tell dbeaver that some values should be NULL in text fields (empty fields works as null only for integers).

So instead of NULL in the script and database, there is '' = empty string. I know there is option Set to NULL on right click, but its not practical because NULL values are randomly positioned in big number of rows.

I have tried NULL, [NULL], empty cell and more and nothing seems to work. 
So my question is, is there a way to "escape" NULL or other way how to tell dbeaver data editor, that these cells from Excel should be NULL after you use Advanced Paste (maybe similar option like for importing CSV)?

Thanks

**Comment by dev-leu** [comment]
This is causing a lot of consternation for my entire team as well.  We routinely work with tables containing forty columns or more, it is a real pain manually sifting through all forty columns of each paste, trying to find all the empty strings that should in fact be nulls and then manually change them one by one.  

Has anyone looked at fixing this yet?  
If not, I am willing to invest my personal time to fix this.  I work as a java application developer, who do I talk to?

**Reply by LonwoLonwo** [reply]
Hello @dev-leu 
What do you want to know?
https://github.com/dbeaver/dbeaver our code is here.
Just make a fork, make a branch, make commits, and do pull requests.

**Comment by LonwoLonwo** [comment]
Hello @radek-vytiska 
You can create a feature request for this problem here: https://github.com/dbeaver/dbeaver/issues

**Comment by radek-vytiska** [comment]
Hello,

@LonwoLonwo thanks for the suggestion, I will check that.

@dev-leu that would be great! Please report back if u manage to find a solution. I hope DBeaver will be able to implement this fix, it would be a huge time saver for us.

**Comment by radek-vytiska** [comment, marked as answer]
Hi again,
I just want to make update on the issue. It seems, that it has been fixed recently: https://github.com/dbeaver/dbeaver/issues/13076

Unfortunately I didnt have opportunity to test it yet, as we are still on older release, but I am really looking forward to it.

Thanks everyone
